taxi/views.py

```diff
@@ -61,16 +63,31 @@
 class CarDetailView(LoginRequiredMixin, generic.DetailView):
     model = Car
 
+    def get_object(self, queryset=None) -> Car:
+        return Car.objects.get(id=self.kwargs["pk"])
+
+    def post(self, request, **kwargs) -> HttpResponse:
+        user = self.request.user
+        car = self.get_object()
+        if user not in car.drivers.all():
+            car.drivers.add(user)
+        else:
+            car.drivers.remove(user)
+
+        return HttpResponseRedirect(reverse(
+            "taxi:car-detail", kwargs={"pk": car.id})
+        )
+
 
 class CarCreateView(LoginRequiredMixin, generic.CreateView):
     model = Car
-    fields = "__all__"
+    form_class = CarForm
     success_url = reverse_lazy("taxi:car-list")
```

Comment:
You should add a `/` at the end of the URLs. Your `success_url` should be `reverse_lazy('taxi:car-list/')`.